### DISCUSSION TITLE
Cost-effectiveness cloud vs physical hardware for mining experiments?

### DISCUSSION THREAD

**Discussion by JDLH** [filecoin-project/community#371] [Mining]
I am curious about Filecoin. Once I have gone through all the tutorials about how to set up a wallet, and store and retrieve data for free or low cost, I would be interested in experimenting with mining. I believe that I would learn a lot about Filecoin and mining by offering actual storage which I would not learn by reading blog posts. 

It looks like I could buy a bunch of PC hardware to make a mining machine, or I could set up a powerful server on some cloud computing platform (AWS, Google Cloud, Azure, etc.) What is the common knowledge about how cost-efficient each of these is?  What are typical costs in North America to mine storage for a year by either of these routes?

And, what are typical revenues for offering storage at this stage of the Filecoin economy?  Is it rare or common that storage miners make a profit on their costs to buy and operate a mining node? 

And — If it is common to make a profit, using a server on a cloud platform, then why haven't all the cloud providers entered the Filecoin storage mining business? :-)

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
- mining hardware in the cloud makes no financial sense
- mining filecoin is not something you just do. at scale it is work

"playing" with mining means that you lock up funds for possibly years. if thats a good thing to "play" with is on you to decide. 

we got the 512MiB testnet back online yesterday, build instructions follow possibly tomorrow - the you could "play" without any rel hardware needs. answer here and i get you up to date as soon as the instructions to build and join are written

**Reply by JDLH** [reply]
> * mining hardware in the cloud makes no financial sense

Well, that's an answer to "what is the common knowledge about how cost-efficient [this] is". Thanks! Has anyone written up the economics behind that statement anywhere?  I'd love to read it.

> * mining filecoin is not something you just do. at scale it is work

Fair enough. But I am interested in learning and understanding. Reading the introductory docs is not deep enough. I don't want to make filecoin mining my fulltime job. I'm looking for a way of learning that is somewhere in the middle.

> "playing" with mining means that you lock up funds for possibly years. if thats a good thing to "play" with is on you to decide.

Interesting! That's part of what I want to learn. It isn't really explained in the introductory documents. Has anyone written up how much funds get locked up, for how many years?

> we got the 512MiB testnet back online yesterday, build instructions follow possibly tomorrow - the you could "play" without any rel hardware needs. answer here and i get you up to date as soon as the instructions to build and join are written

That sounds interesting. Please sign me up!

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
> Well, that's an answer to "what is the common knowledge about how cost-efficient [this] is". Thanks! Has anyone written up the economics behind that statement anywhere? I'd love to read it.

i don't think so. cloud hardware is expensive. you'll need a lot of calculation power, quite exotic one in addition (sha-ni instruction set and nvidia gpus). as you are in it for at least a year, if doing it reasonable, even 2 years (due to locked up FIL) you are most likely way better off with your own hardware. and it is not about mining but providing storage in the long term. all you do now is just building a foundation for the future to become a storage provider using filecoin.

>  Fair enough. But I am interested in learning and understanding. Reading the introductory docs is not deep enough. I don't want to make filecoin mining my fulltime job. I'm looking for a way of learning that is somewhere in the middle.

testnets are an option. calibrationnet needs mainnet hardware, the one we offer needs basically no hardware at all. i'll let you knwo the details. as soon as you are on mainnet the stakes and the risk are high. 

> Interesting! That's part of what I want to learn. It isn't really explained in the introductory documents. Has anyone written up how much funds get locked up, for how many years?

50 FIL per TiB right now. at least 6 month. with client data deals most likely 18 month. it's fluid and depends on the network economics. see filfox.info for JIT values (pledge collateral is what you lock up). pledge collateral grants you storage power and storage power allows you to mine blocks. so you lock up FIL or do not mine. simple as it is. you need 10TiB storage power to get rewards, 500FIL - so there is no "play" around on mainnet.
